vms/platformvm/state/state.go

```diff
@@ -2467,61 +2483,40 @@ func (s *state) calculateValidatorDiffs() (map[subnetIDNodeID]*validatorDiff, er
 		}
 	}
 
-	// Perform SoV deletions:
-	for validationID := range s.sovDiff.modified {
+	// Calculate the changes to the ACP-77 validator set
+	for validationID, sov := range s.sovDiff.modified {
 		priorSOV, err := s.getPersistedSubnetOnlyValidator(validationID)
-		if err == database.ErrNotFound {
-			continue
-		}
-		if err != nil {
-			return nil, err
-		}
-
-		var (
-			diff           *validatorDiff
-			subnetIDNodeID = subnetIDNodeID{
+		if err == nil {
+			// Delete the prior validator
+			subnetIDNodeID := subnetIDNodeID{
 				subnetID: priorSOV.SubnetID,
+				nodeID:   priorSOV.effectiveNodeID(),
 			}
-		)
-		if priorSOV.isActive() {
-			subnetIDNodeID.nodeID = priorSOV.NodeID
-			diff = getOrDefault(changes, subnetIDNodeID)
-			diff.prevPublicKey = priorSOV.PublicKey
-		} else {
-			subnetIDNodeID.nodeID = ids.EmptyNodeID
-			diff = getOrDefault(changes, subnetIDNodeID)
+			diff := getOrDefault(changes, subnetIDNodeID)
+			if err := diff.weightDiff.Add(true, priorSOV.Weight); err != nil {
```

Comment:
not directly related with this PR but can we use `weightDiff.Remove(amount)` rather then `weightDiff.Add(true, amount)`?

Reply:
Refactored into #3514